src/main/java/com/Model/User.java

```diff
@@ -1,57 +1,78 @@
 package com.Model;
 
-import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.userdetails.UserDetails;
 import javax.persistence.*;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
 
 @Entity
-@Inheritance (strategy = InheritanceType.JOINED)
-public class User {
+public class User implements UserDetails {
 
     @Id
     @GeneratedValue(strategy = GenerationType.AUTO)
     private long id;
+
     private String emailAddress;
     private String password; //Encrypted password
-    private boolean isEnabled; //False if user has not confirmed email address yet
-    private boolean isVerified;
 
-    @Autowired
     public User(){
         this("unknown", "unknown");
     }
 
     public User (String emailAddress, String password){
         this.emailAddress = emailAddress;
         this.password = password;
-        this.isEnabled = false;
-        this.isVerified = false;
     }
 
     public String getEmailAddress (){
         return this.emailAddress;
     }
 
+    @Override
+    public Collection<? extends GrantedAuthority> getAuthorities() {
+        List<GrantedAuthority> roles = new ArrayList<>();
+        roles.add(new SimpleGrantedAuthority("USER"));
+        return roles ;
+    }
+
     public String getPassword (){
         return this.password;
     }
 
-    public void setPassword (String password){
-        this.password = password;
+    @Override
+    public String getUsername() {
+        return this.emailAddress;
     }
 
-    public long getId (){
-        return this.id;
+    @Override
+    public boolean isAccountNonExpired() {
```

Comment:
if this is always true, why do we need it?

Reply:
This class is implementing UserDetails Interface from Spring Framework, UserDetailService uses this class when user logs in. For now just return true, but in the future, we can add more account management functionalities to user accounts with those methods.